module-2/routes/users.ts

```diff
@@ -0,0 +1,84 @@
+import express from 'express';
+import { v4 as uuid } from 'uuid';
+import User from '../models/User';
+import { Users } from '../db';
+import schema from '../schema';
+import { filterNotDeleted, getAutoSuggestUsers } from '../utils';
+
+const router = express.Router();
+
+router.get('/', (req, res) => {
+  res.send([...Users.values()].filter(filterNotDeleted));
+});
+
+router.get('/all', (req, res) => {
+  res.send([...Users.values()]);
+});
+
+const DEFAULT_LIMIT = 3;
+router.get('/autosuggest', (req, res) => {
+  const { login = '', limit = DEFAULT_LIMIT } = req.query;
+  res.send(getAutoSuggestUsers(Users)(String(login), Number(limit)));
+});
+
+router.get('/:id', (req, res) => {
+  const user = Users.get(req.params.id);
+  !user?.isDeleted
+    ? res.send(user)
+    : res.status(404).send({ message: 'User not found!' });
+});
+
+router.post('/', (req, res) => {
+  const { login, password, age } = <User>req.body;
+  const { error } = schema.validate({ login, password, age });
```

Comment:
Consider creating a separate middleware for such purposes as it is a more common practise.
https://dev.to/itnext/joi-awesome-code-validation-for-node-js-and-express-35pk

Reply:
Did that, thanks